### ISSUE TITLE
Possibility to use 2-Factor Authentication configured per user

### ISSUE THREAD

**Issue by toskrip** [reliatec-gmbh/LibreClinica#286]
**Description:**
The idea is to allow users to switch from standard password based authentication to 2-factor authentication (with second factor generated by one time password algorithm). The initial setup should be supported with on display QR code scan (application) or PDF QR code printout letter. System wide settings to enforce the 2-Factor for all users after specific date.

**Requirements:**
Basic installation with at least one study and one user setup with 2FA (compatible one time password application) configured with system wide settings to allow 2-FA (section 23 2-Factor Authentication (2-FA) in datainfo.properties file).

**Steps to follow:**
1. enter Username and Password into login form
1. open configured one time password application to generate OTP
1. enter OTP into 2nd Factor
1. login

**Expected result:**
If authentication succeeded  and OTP is valid login into application otherwise bad user credentials.

**Additional context (optional):**
More details in test specifications: T001-08, T001-09, T001-10, T003-09, T003-10, T003-11, T038-01, T038-02

**Comment by toskrip** [comment]
Is implemented in lc-develop-2fauth branch. Issue will be auto closed once the branch is merged into lc-develop.
